go/teams/seitan_v2.go

```diff
@@ -0,0 +1,201 @@
+package teams
+
+import (
+	"bytes"
+	"fmt"
+	"strings"
+
+	"crypto/hmac"
+	"crypto/rand"
+	"crypto/sha512"
+	"encoding/base64"
+
+	"github.com/keybase/client/go/libkb"
+	"github.com/keybase/go-crypto/ed25519"
+	"golang.org/x/net/context"
+
+	keybase1 "github.com/keybase/client/go/protocol/keybase1"
+)
+
+// Seitan tokens v2 have a '+' as the sixth character. We use this
+// to distinguish from email invite tokens (and team names).
+const seitanEncodedIKeyV2PlusOffset = 6
+
+// "Invite Key Version 2"
+type SeitanIKeyV2 string
+
+func GenerateIKeyV2() (ikey SeitanIKeyV2, err error) {
+	str, err := generateIKey(seitanEncodedIKeyV2PlusOffset)
+	if err != nil {
+		return ikey, err
+	}
+	return SeitanIKeyV2(str), err
+}
+
+// GenerateIKeyV2FromString safely creates SeitanIKey value from
```

Comment:
think function doesn't seem accurately named.  It seems like `Parse` rather than `Generate`

Reply:
done